core/src/test/java/hudson/GetLocaleStaticHelpUrlTest.java

```diff
@@ -0,0 +1,220 @@
+        );
+
+        URL id = Descriptor.getStaticHelpUrl(req, klass, "-id");
+        localeResourceIs(id, "help-id.html");
+    }
+
+    @Test
+    public void getStaticHelpUrlAcceptZhResZhCountry() {
+        // Accept-Language: zh,zh-CN;q=0.9,en-US;q=0.8,en;q=0.7,zh-TW;q=0.6
+        StaplerRequest req = mockStaplerRequest(
+                Locale.CHINESE,
+                Locale.SIMPLIFIED_CHINESE,
+                Locale.ENGLISH,
+                Locale.US,
+                Locale.TRADITIONAL_CHINESE
+        );
+
+        Klass klass = mockKlass(
+                "help-id.html",
+                "help-id_zh_CN.html"
+        );
+
+        URL id = Descriptor.getStaticHelpUrl(req, klass, "-id");
+        localeResourceIs(id, "help-id_zh_CN.html");
+    }
+
+    @Test
+    public void getStaticHelpUrlAcceptZhResBoth() {
+        // Accept-Language: zh,zh-CN;q=0.9,en-US;q=0.8,en;q=0.7,zh-TW;q=0.6
+        StaplerRequest req = mockStaplerRequest(
+                Locale.CHINESE,
+                Locale.SIMPLIFIED_CHINESE,
+                Locale.ENGLISH,
+                Locale.US,
+                Locale.TRADITIONAL_CHINESE
+        );
+
+        Klass klass = mockKlass(
+                "help-id.html",
+                "help-id_zh.html",
+                "help-id_zh_CN.html"
+        );
+        URL id = Descriptor.getStaticHelpUrl(req, klass, "-id");
+        localeResourceIs(id, "help-id_zh.html");
+    }
+
+    @Test
+    public void getStaticHelpUrlAcceptZhResMore() {
+        // Accept-Language: zh,zh-CN;q=0.9,en-US;q=0.8,en;q=0.7,zh-TW;q=0.6
+        StaplerRequest req = mockStaplerRequest(
+                Locale.CHINESE,
+                Locale.SIMPLIFIED_CHINESE,
+                Locale.ENGLISH,
+                Locale.US,
+                Locale.TRADITIONAL_CHINESE
+        );
+
+        Klass klass = mockKlass(
+                "help-id.html",
+                "help-id_zh_CN.html",
+                "help-id_zh_TW.html",
+                "help-id_de_DE.html"
+        );
+        URL id = Descriptor.getStaticHelpUrl(req, klass, "-id");
+        localeResourceIs(id, "help-id_zh_CN.html");
+    }
+
+    private StaplerRequest mockStaplerRequest(Locale... localeArr) {
+        StaplerRequest req = mock(StaplerRequest.class);
+        Enumeration<Locale> locales = Collections.enumeration(Arrays.asList(localeArr));
+        when(req.getLocales()).thenReturn(locales);
+        return req;
+    }
+
+    private Klass mockKlass(String... resources) {
+        Klass klass = mock(Klass.class);
+        for (String resource : resources) {
+            when(klass.getResource(resource)).thenReturn(getUrl("https://jenkins/" + resource));
+        }
+        return klass;
+    }
+
+    private URL getUrl(final String realUrl) {
+        try {
+            return new URL(realUrl);
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private void localeResourceIs(URL url, String resource) {
```

Comment:
one nit I would add assert to the beginning of this method:
```suggestion
    private void assertThatLocaleResourceIs(URL url, String resource) {
```